parse_lcsh_file.rb

```diff
@@ -0,0 +1,37 @@
+def find_subject(pref_label_json)
+  if pref_label_json.is_a? Hash
+    subject = pref_label_json["@value"]
+  elsif pref_label_json.is_a? Array
+    subject = pref_label_json.find{ |v| v.is_a? String }
+  elsif pref_label_json.is_a? String
+    subject = pref_label_json
+  else
+    # if you end up here you are probably going to want to see why
+    # binding.pry
+    print pref_label_json
+    raise "#find_subject found an unexpected value, stopping here"
+  end
+
+  subject&.gsub('"', '\\"')
+end
+
+# lcsh subject headings with SKOS/RDF formatted into ndjson
+lcsh_file = open("lcsh.skos.ndjson")
+lcsh_yaml_file = open("subjects_lcsh.yml", "w")
+memoized_ids = Set.new
+
+lcsh_file.each_line do |line|
+  line_graph_json = JSON(line)["@graph"]
+
+  line_graph_json.each do |node|
+    if node["@type"] == "skos:Concept" && node["skos:prefLabel"].present?
+      id = node["@id"]
+
+      if !memoized_ids.include?(id)
+        memoized_ids << id
+        lcsh_string = %(- id: #{id}\n  scheme: lcsh\n  subject: "#{find_subject(node["skos:prefLabel"])}"\n)
```

Comment:
InvenioRDM double-dips on the scheme and uses it as a human readable label too so we want it cased appropriately.

```suggestion
        lcsh_string = %(- id: #{id}\n  scheme: LCSH\n  subject: "#{find_subject(node["skos:prefLabel"])}"\n)
```

On another note, I guess the stdlib YAML would have wanted the whole document in memory before dumping it (https://ruby-doc.org/stdlib-2.5.1/libdoc/yaml/rdoc/YAML.html). This may or may not have been ideal given the size of data we are dealing with. Alternative would have been to use the YAML stream writing and remove the "---" after the fact. Messy either way. Even in Python I haven't found a decent YAML array streaming library...  This works, so :+1: